File: py/trello_sdk/features.py
# Trello SDK feature factory

from trello_sdk.feature.base_feature import TrelloBaseFeature
from trello_sdk.feature.debug_feature import TrelloDebugFeature
from trello_sdk.feature.idempotency_feature import TrelloIdempotencyFeature
from trello_sdk.feature.metrics_feature import TrelloMetricsFeature
from trello_sdk.feature.paging_feature import TrelloPagingFeature
from trello_sdk.feature.ratelimit_feature import TrelloRatelimitFeature
from trello_sdk.feature.retry_feature import TrelloRetryFeature
from trello_sdk.feature.test_feature import TrelloTestFeature
from trello_sdk.feature.timeout_feature import TrelloTimeoutFeature


_FEATURES = {
    "base": lambda: TrelloBaseFeature(),
    "debug": lambda: TrelloDebugFeature(),
    "idempotency": lambda: TrelloIdempotencyFeature(),
    "metrics": lambda: TrelloMetricsFeature(),
    "paging": lambda: TrelloPagingFeature(),
    "ratelimit": lambda: TrelloRatelimitFeature(),
    "retry": lambda: TrelloRetryFeature(),
    "test": lambda: TrelloTestFeature(),
    "timeout": lambda: TrelloTimeoutFeature(),
}


def _make_feature(name):
    factory = _FEATURES.get(name)
    if factory is not None:
        return factory()
    return _FEATURES["base"]()


# True when this SDK was generated with the named feature class - the
# constructor's tolerance for extend-carried features reads this (an
# active name with no generated class must not become a BaseFeature
# stray when an extend instance carries it).
def _has_feature(name):
    return name in _FEATURES

File: php/features.php
<?php
declare(strict_types=1);

// Trello SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/DebugFeature.php';
require_once __DIR__ . '/feature/IdempotencyFeature.php';
require_once __DIR__ . '/feature/MetricsFeature.php';
require_once __DIR__ . '/feature/PagingFeature.php';
require_once __DIR__ . '/feature/RatelimitFeature.php';
require_once __DIR__ . '/feature/RetryFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';
require_once __DIR__ . '/feature/TimeoutFeature.php';


class TrelloFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new TrelloBaseFeature();
            case "debug":
                return new TrelloDebugFeature();
            case "idempotency":
                return new TrelloIdempotencyFeature();
            case "metrics":
                return new TrelloMetricsFeature();
            case "paging":
                return new TrelloPagingFeature();
            case "ratelimit":
                return new TrelloRatelimitFeature();
            case "retry":
                return new TrelloRetryFeature();
            case "test":
                return new TrelloTestFeature();
            case "timeout":
                return new TrelloTimeoutFeature();
            default:
                return new TrelloBaseFeature();
        }
    }

    /**
     * Does a generated feature class back this name? False for a name only
     * an options extend instance can supply (the station adopt path) - the
     * constructor uses this to skip make_feature for such names instead of
     * adding a stray BaseFeature.
     */
    public static function has_feature(string $name): bool
    {
        switch ($name) {
            case "base":
            case "debug":
            case "idempotency":
            case "metrics":
            case "paging":
            case "ratelimit":
            case "retry":
            case "test":
            case "timeout":
                return true;
            default:
                return false;
        }
    }
}
